app/models/host/managed.rb

```diff
@@ -926,6 +927,17 @@ def apply_compute_profile(modification)
 
   private
 
+  def owner_taxonomies_match
+    if self.owner.is_a?(User)
+      if Organization.organizations_enabled && self.organization_id && !self.owner.my_organizations.where(:id => self.organization_id).exists?
```

Comment:
I agree, I created http://projects.theforeman.org/issues/16128 to track this (and volunteer to do it). I hope that does not block this PR since it's unrelated change. These methods were added long time ago.